### ISSUE TITLE
pressing tab gives me only root commands, even when rott command already typed

### ISSUE THREAD

**Issue by remort** [pallets/click#721]
I have subcomands and sub-subcommands in my cli app. But root groups added with `cli.add_command` in the top module where cli() is defined.

Bash completion script is generated with `eval "$(_MYCLI_COMPLETE=source myclil)";` successfully. But it is suspiciously short.

Bash completion is based on this `cli.add_command` as i get it, but I can't understand how to enable subcommands be shown in shell when i hit tab button.

And as I mentioned above, top root commands shouldn't be shown since one of them is already typed in.

Documentation is silent about that, so please provide some guide on how to create a complex bash complpetion for cli applications made with click.

**Comment by untitaker** [comment]
Please [give a minimal example application that doesn't work for you](http://www.chiark.greenend.org.uk/~sgtatham/bugs.html)

**Comment by remort** [comment]
cli.py
```
import click

from ideco_cli.commands.users import users
from ideco_cli.commands.sample import sample


@click.group()
def cli() -> None:
    pass

cli.add_command(sample)


if __name__ == '__main__':
    cli()
```

sample module under commands catalog (no __init__.py inside a catalog):
```
import click


@click.group()
@click.help_option()
def sample():
    pass


@sample.command()
@click.argument('mode', nargs=1, required=False, type=click.STRING)
def access(mode):
    click.exho(mode)


@sample.command()
@click.option('--password', prompt=True, hide_input=True, confirmation_prompt=True)
def permissions(password):
    print(password)

@sample.group()
@click.help_option()
def sub_sample():
    pass

@sub_sample.command()
def set():
    pass

@sub_sample.command()
def get():
    pass
```

executable is in a place:
```
# which ics
/usr/bin/ics
```

generate bashcompletion file:
```
eval "$(_ICS_COMPLETE=source ics)"
```

Everything is set now.

Bash completion in shell pressing tabs:

```
[root@unknown-external-domain ideco_cli]# ics sample sample 
access       permissions  sub_sample   
[root@unknown-external-domain ideco_cli]# ics sample sample 
access       permissions  sub_sample   
[root@unknown-external-domain ideco_cli]# ics sample sample sub_sample 
access       permissions  sub_sample   
[root@unknown-external-domain ideco_cli]# ics sample sample sub_sample 
access       permissions  sub_sample   
[root@unknown-external-domain ideco_cli]# ics sample sample sub_sample sub_sample 
get  set  
[root@unknown-external-domain ideco_cli]# ics sample sample sub_sample sub_sample get 
get  set  
[root@unknown-external-domain ideco_cli]# ics sample sample sub_sample sub_sample get get 
cli.py           cli.py-bk        commands/        __init__.py      __pycache__/     rest_request.py  tests/ 
```

So it shows root and group commands twice. After typing them, completion goes further. Seems good, except doubling root and group commands. How I can avoid this ?

**Comment by remort** [comment]
The completion file itself:
```
_ics_completion() {
    COMPREPLY=( $( env COMP_WORDS="${COMP_WORDS[*]}" \
                   COMP_CWORD=$COMP_CWORD \
                   _ICS_COMPLETE=complete $1 ) )
    return 0
}

complete -F _ics_completion -o default ics;
```

**Comment by remort** [comment]
Dealing with context (`ctx`) seems to have no effect.

**Comment by davidism** [comment]
From user dlovell on Discord:

> I am not able to recreate the problem. The only difference between what I have that works and what they show that doesn't is that their executable is in `/usr/bin` instead of a virtualenv. Their autogenerated completion file looks different in form as well. Even modifying the completion file to look like theirs does not regenerate the issue.

Going to close this for now as it may have been addressed by other changes to the autocomplete system. If you're still having this issue, let us know.

**Comment by dlovell** [comment]
**tl;dr: recommend closing this as no longer an issue in click=7.0**

Using click=7.0, I am not able to reproduce this issue.

I've created a [gist](https://gist.github.com/dlovell/0357739ac6c9a6220208de909bc4bc6a) that tries to replicate @remort 's files and description of the problem (where [github_issue_721-catalog-sample.py](https://gist.github.com/dlovell/0357739ac6c9a6220208de909bc4bc6a#file-github_issue_721-catalog-sample-py) should be `github_issue_721/catalog/sample.py`).

One notable difference is that @remort has the binary in /usr/bin whereas in the gist the executable will be in the virtualenv.

[transcript.txt](https://gist.github.com/dlovell/0357739ac6c9a6220208de909bc4bc6a#file-transcript-txt) shows the behavior with the completion script as generated by click=7.0 as well as trying to replicate what @remort has for it.  When I run a similar completion script, I **do** get the behavior where a tab with no further completions shows files.  Perhaps the issue existed at one time, but no longer does.
